core/translate.c

```diff
@@ -387,16 +418,58 @@ translate_walk_good_state(dcontext_t *tdcontext, translate_walk_t *walk,
                           app_pc translate_pc)
 {
     return (!walk->unsupported_mangle ||
-            /* If we're at the instr AFTER the mangle region, we're ok */
+            /* If we're at the instr AFTER the mangle region, or at an instruction
+             * in the mangled region's EPILOGUE, we're ok. If it's epilogue, we only
+             * support a simple spill/restore case, which is checked in
+             * translate_walk_restore.
```

Comment:
For the supported simple case, it will be walk->in_mangle_region && translate_pc != walk->translation

walk->translation will be the next PC due to being the mangling epilogue. This only holds for the simplest case, hence the comment. Unclear?

Reply:
(Re-open if necessary)

Reply:
My point is that even if you're in the epilogue you still need to *not* have seen an unsupported mangle instr -- at least when using rollback state.  So you would never get to this second condition. (If you rewrote it to use rollforward state that may change.)

Reply:
Ok, removed comment.